### DISCUSSION TITLE
useRequest用不了staleTime，throttleWait等？

### DISCUSSION THREAD

**Discussion by weeptao** [umijs/umi#11597] [Q&A]
我使用的是umimax@4，useRequest是内置的ahooks@2版本
![image](https://github.com/umijs/umi/assets/103228833/7efed60a-d6c8-4157-b6ff-e6bfc817751f)
![image](https://github.com/umijs/umi/assets/103228833/86a70f42-0ac7-4370-8c12-cd895e7d6a30)
但是这个不管是节流还是缓存，好像都用不了。控制台也没有任何报错与警告。是我的方式不对吗

**Comment by fz6m** [comment]
umi 4 里的 useRequest 是为了方便 umi 3 升级上来所以内置的 ahooks use request 版本没变还是旧的，目前已经不推荐使用 dva / use request 这些过时、很久不再维护的方案。

现代数据流更推荐底层自己封装 axios/ fetch ，顶层使用 [react-query](https://umijs.org/docs/max/react-query) / [swr](https://github.com/vercel/swr) 管理请求状态，[valtio](https://umijs.org/docs/max/valtio) / [jotai](https://github.com/pmndrs/jotai) 管理全局状态。

**Reply by DAOLIDEWONIU** [reply]
有没有实践demo ？
